doc/stake-pool-operations/KES_period.md

```diff
@@ -4,43 +4,185 @@ To create an operational certificate for a block-producing node, you need a _KES
+
+```
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 4,
+```
+
+### My pool has not forged a block in the current KES interval and I need to issue new operational certificate. What should I do with the counter?
+
+Keep in mind that the rule is that the counter must be one more than the last counter used to _actually_ forge a block in the _current chain_. So if we have not forged a block during the current KES interval, the counter must not be incremented when issuing a new OpCert i.e it remains the same.
+
+In this scenario `cardano-cli query kes-period-info` will show that the **OnDisk** counter is greater than the **On-Chain** counter
+
+```
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 3,
+```
+
+Therefore, we **cannot** use `cardano-cli node issue-op-cert...` right away to issue a new operational certificate because this would increment the **OnDisk** counter by 1 more, resulting in something like:
+
+```
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 4,
+```
+
+If we issue a certificate with such a counter, any block we forge with it will be an **Invalid Block**. The node logs will show the **error**:
+
+```
+[CLR:cardano.node.ChainDB:Error:25] [2022-06-30 03:25:37.01 UTC] Invalid block 237602735e6b56985109480aefbc4821f57e6389736be238ffbec4c0188f9702 at slot 5206: ExtValidationErrorHeader (HeaderProtocolError (HardForkValidationErrFromEra S (S (S (S (S (Z (WrapValidationErr {unwrapValidationErr = CounterOverIncrementedOCERT 2 4}))))))))
+```
+
+### Recovering after having issued an _invalid_ certificate OR to issue a new certificate when our pool has not forged a block during the current KES interval:
+
+1. Query `kes-period-info`. You can save the json output to a file using the `--out-file` flag
+
+```bash
+> cardano-cli query kes-period-info --testnet-magic 42 --op-cert-file node-spo3/opcert.cert --out-file kes_period_info.json
+
+> cat kes_period_info.json
 
-    {
-        "epoch": 259,
-        "hash": "dbf5104ab91a7a0b405353ad31760b52b2703098ec17185bdd7ff1800bb61aca",
-        "slot": 26633911,
-        "block": 5580350
-    }
+>
+{
+    "qKesCurrentKesPeriod": 18,
+    "qKesEndKesInterval": 18,
+    "qKesKesKeyExpiry": null,
+    "qKesMaxKESEvolutions": 6,
+    "qKesNodeStateOperationalCertificateNumber": 2,
+    "qKesOnDiskOperationalCertificateNumber": 4,
+    "qKesRemainingSlotsInKesPeriod": 198,
+    "qKesSlotsPerKesPeriod": 300,
+    "qKesStartKesInterval": 12
+}
+```
+2. Create a new counter (on your cold environment)
 
-In this example, we are currently in slot 26633911, and we know from the genesis file that one period lasts for 129600 slots. So we calculate the current period by
+Take the **OnChain** counter from above. In this case `"qKesNodeStateOperationalCertificateNumber": 2` and increment it by exactly 1 when issuing the new counter:
```

Comment:
I think the the values in your json blob are mixed up. They should be
```
    "qKesNodeStateOperationalCertificateNumber": 2,
    "qKesOnDiskOperationalCertificateNumber": 2,
 ```
 also
 
 "In this case `"qKesOnDiskOperationalCertificateNumber": 2` and increment it by exactly 1 when issuing the new counter"

Reply:
This section describes how to recover after having issued an invalid certificate (wrong counter value) or issuing a new certificate when the pool has not forged a block on the current KES interval.  So the json is an example of how kes-period-info looks like when it is wrong.